### PR TITLE
Add support for signed GET requests for aws authentication

### PR DESCRIPTION
**Pull request by bluestealth**
Adds support for signed GET requests of GetCallerIdentity for authenticating to vault using the aws iam method.

This is related to https://github.com/aws/aws-sdk-go-v2/issues/1137

### CODE REVIEW

**Comment by GromNaN** [comment]
👍🏻 Thank you for working on it. Supporting presigned GET requests is a valuable feature to make integration easier. AWS SDKs offers method to presign requests, but we have to use "internals" in order to catch method/path/headers/body of the sts request.

**Comment by GromNaN** [comment]
Hello, is there any security concern before adding this feature?

File: builtin/credential/aws/path_login_test.go [hunk 1]
```diff
@@ -121,9 +121,110 @@ func TestBackend_pathLogin_parseIamArn(t *testing.T) {
...
+	})
+	getQueryInvalidAction := url.Values(map[string][]string{
+		"X-Amz-Algorithm":     {"AWS4-HMAC-SHA256"},
+		"X-Amz-Credential":    {"AKIDEXAMPLE/20150830/us-east-1/iam/aws4_request"},
+		"X-Amz-SignedHeaders": {"host;x-vault-aws-iam-server-id"},
+		"X-Amz-Signature":     {"5d672d79c15b13162d9279b0855cfba6789a8edb4c82c400e06b5924a6f2b5d7"},
+		"X-Amz-User-Agent":    {"aws-sdk-go-v2/1.2.0 os/linux lang/go/1.16 md/GOOS/linux md/GOARCH/amd64"},
+		"Action":              {"GetSessionToken"},
+		"Version":             {"2011-06-15"},
+	})
```
Comment (maxcoulombe) [inline builtin/credential/aws/path_login_test.go:172]: I think there is some coverage missing for multiple actions.

**Comment by maxcoulombe** [comment]
Hey @bluestealth, thanks a lot for your work on this. Kudos for thinking about the tests and docs as well.

The code looks sound, my only 2 comments are really minor nitpicking. Do you think it'd be possible to update the branch to resolve the conflict and add a [changelog entry](https://github.com/hashicorp/vault/tree/main/changelog) so I can have one last look that everything is in order before approving?

The changelog could be something like:

10961.txt
```
```release-note:improvement
auth/aws: Added support for signed GET requests for authenticating to vault using the aws iam method.
```
```

File: builtin/credential/aws/path_config_client.go [hunk 2]
```diff
@@ -349,6 +350,9 @@ type clientConfig struct {
 func (c *clientConfig) validateAllowedSTSHeaderValues(headers http.Header) error {
 	for k := range headers {
 		h := textproto.CanonicalMIMEHeaderKey(k)
+		if h == "X-Amz-Signedheaders" {
+			h = "X-Amz-SignedHeaders"
+		}
```
[the remark this answers is not in the document]
Comment (bluestealth) [reply builtin/credential/aws/path_config_client.go:355]: Good Point 👍🏻

File: builtin/credential/aws/path_login_test.go (hunk 1 above)
Comment (bluestealth) [reply builtin/credential/aws/path_login_test.go:172]: Good catch, I missed that case.

**Review by maxcoulombe — APPROVED** [review]
Looks good, thanks for the modification & adding the test case.

Don't worry about the Vercel step, from what I understand it does not work with forks. I validated the doc changes with `cd website && make website` and it's ok.

**Comment by maxcoulombe** [comment]
This completely slipped under my radar apologies for the delay and thanks for that contribution!

**Pull request merged**